src/WpfMath/TexFormulaParser.cs

```diff
@@ -273,6 +284,55 @@ private SourceSpan ReadGroup(TexFormula formula, SourceSpan value, ref int posit
             position++;
             return value.Segment(start, position - start - 1);
         }
+        
+        private string ReadGroup(string str, char leftchar, char rightchar, int startPosition)
```

Comment:
This method uses a wrong approach to markup parsing.

We usually don't copy the original string, but operate on `SourceSpan`s that all point to the original string. Probably we could strip this method and use an existing `ReadGroup` instead.